### PR TITLE
install firewall rules with installer

### PR DESCRIPTION
**Pull request by Laserlicht**
no more visible firewall window on windows...

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -713,12 +713,22 @@ if(WIN32)
 	endif()
 	if(ENABLE_LAUNCHER)
 		set(CPACK_PACKAGE_EXECUTABLES "VCMI_launcher;VCMI")
-		set(CPACK_NSIS_EXTRA_INSTALL_COMMANDS " CreateShortCut \\\"$DESKTOP\\\\VCMI.lnk\\\" \\\"$INSTDIR\\\\VCMI_launcher.exe\\\"")
+		set(CPACK_NSIS_EXTRA_INSTALL_COMMANDS "
+			CreateShortCut \\\"$DESKTOP\\\\VCMI.lnk\\\" \\\"$INSTDIR\\\\VCMI_launcher.exe\\\"
```
Comment (kambala-decapitator) [inline CMakeLists.txt:717]: make the exe name a param, then you can greatly simplify the if/else

Comment (Laserlicht) [reply CMakeLists.txt:717]: How to do this in cmake? simple do do a set before and use $variablename?

**Comment by dydzio0614** [comment]
my take: merge as is and do not enforce people unfamiliar with cmake to refactor stuff on their own

**Comment by Laserlicht** [comment]
> my take: merge as is and do not enforce people unfamiliar with cmake to refactor stuff on their own

Yes would properly the be best. :)

Can optimized directly afterwards by a cmake person. :)

File: CMakeLists.txt (hunk 1 above)
Comment (kambala-decapitator) [reply CMakeLists.txt:717]: if you don't want to read cmake docs / some tutorial (although I think it's a good opportunity to read some basics), there's a bunch of such examples in our CML. you declare variables with `set(varname value)` and use it inside a string as `${varname}`

**Comment by kambala-decapitator** [comment]
> my take: merge as is and do not enforce people unfamiliar with cmake to refactor stuff on their own

IMO such attitude makes code review senseless

File: CMakeLists.txt (hunk 1 above)
Comment (Laserlicht) [reply CMakeLists.txt]: You mean like this?

Comment (kambala-decapitator) [reply CMakeLists.txt]: yep, looks right. Well done!

**Comment by kambala-decapitator** [comment]
btw have you tested this solution?

**Comment by Laserlicht** [comment]
Yes. Installing and uninstalling works. Tested with Win11.

**Review by kambala-decapitator — APPROVED** [review]

**Comment by Nordsoft91** [comment]
Relevant to https://github.com/vcmi/vcmi/issues/2999 ?

**Comment by Alexander-Wilms** [comment]
Only the MinGW .exe is flagged as malicious:

mingw exe: https://www.virustotal.com/gui/file/15f97d3c88a2d275e82a2745daba31baf9339f8ddefbed541895b6a6033b8fcd?nocache=1
mingw zip: https://www.virustotal.com/gui/file/d7e9de16cffac41ba213d1fe39329eba3db3822af55d4058f1d154324fa83900?nocache=1
msvc exe: https://www.virustotal.com/gui/file/f33ed588beed51ca30142c6e3ce9deefc295c905bfc32c9141f9f33f381b0e54?nocache=1
msvc zip: https://www.virustotal.com/gui/file/7e36c43d824cc9f511e8df43ad49c1931ec5354c75542045e5b9a6b2eff32bc4?nocache=1

**Comment by Laserlicht** [comment]
> Only the MinGW .exe is flagged as malicious:

That's better than 1.3.2.

I think there are many installer modifying the firewall.

**Pull request merged**